### PR TITLE
recover from job panic

### PR DESCRIPTION
**Pull request by shz117**
so it doesn't crash the server.

### CODE REVIEW

File: memstore/scheduler.go [hunk 1]
```diff
@@ -228,6 +228,12 @@ func (scheduler *schedulerImpl) Start() {
 }
 
 func (scheduler *schedulerImpl) executeJob(jb *jobBundle) {
+	defer func() {
```
Comment (lucafuji) [inline memstore/scheduler.go]: IMO better to crash the server to catch the bug

File: memstore/scheduler.go [hunk 2]
```diff
@@ -228,6 +228,12 @@ func (scheduler *schedulerImpl) Start() {
...
+		if r := recover(); r != nil {
+			utils.GetLogger().Errorf("Job failed a fter panic: %v", r)
```
Comment (lucafuji) [inline memstore/scheduler.go]: use log fields and there is a typo

Comment (lucafuji) [reply memstore/scheduler.go]: And better to treat the error differently according to its type

File: memstore/scheduler.go (hunk 1 above)
Comment (shz117) [reply memstore/scheduler.go]: crashing server means downtime, can't be good?

we can add metrics to discover job failure, plus logs to investigate.

File: memstore/scheduler.go [hunk 3]
```diff
@@ -228,8 +228,15 @@ func (scheduler *schedulerImpl) Start() {
 }
 
 func (scheduler *schedulerImpl) executeJob(jb *jobBundle) {
+	defer func() {
+		if r := recover(); r != nil {
+			utils.GetRootReporter().GetCounter(utils.JobFailuresCount).Inc(1)
+			utils.GetLogger().With("error", r).Error("Job failed after panic")
```
Comment (lucafuji) [inline memstore/scheduler.go]: better to have table, shard and job type information

**Pull request merged**